hkmc2/shared/src/main/scala/hkmc2/semantics/ucs/Translator.scala

```diff
@@ -137,19 +116,31 @@ class Translator(val elaborator: Elaborator)
           inner(captures2 ++ captures1, postfixScrut2)))
     case Under() => inner(Map.empty, scrut) // TODO: check if this is correct
     case ctor @ (_: Ident | _: Sel) =>
-      val clsTrm = elaborator.cls(elaborator.term(ctor), inAppPrefix = false)
-      clsTrm.symbol.flatMap(_.asClsLike) match
-      case S(cls: (ClassSymbol | ModuleSymbol)) =>
-        val kind = cls match { case _: ClassSymbol => "class" case _ => "module" }
-        error(msg"Cannot treat this $kind as a string prefix" -> ctor.toLoc)
-        errorSplit
-      case S(psym: PatternSymbol) =>
-        makeUnapplyStringPrefixBranch(scrut(), clsTrm, postfixSym =>
-          inner(Map.empty, () => postfixSym.ref())
-        )(Split.End)
-      case _ =>
-        error(msg"Cannot use this ${ctor.describe} as an extractor" -> ctor.toLoc)
-        errorSplit
+      val ctorTrm = term(ctor)
+      val prefixSymbol = new TempSymbol(N, "prefix")
+      val postfixSymbol = new TempSymbol(N, "postfix")
+      val mode = MatchMode.StringPrefix(prefixSymbol, postfixSymbol)
+      val pattern = Pattern.ClassLike(ctorTrm, N, mode, false)(ctor)
+      Branch(scrut(), pattern, inner(Map.empty, () => postfixSymbol.ref())) ~: Split.End
+      
+      // Because we don't resolve the symbol of `ctor` here, we don't know if
+      // it is a class, a module, or a pattern symbol. We no longer need to
+      // make the unapply branch here and it will be done in normalization.
+      // - - - - - //
+      // OLD CODE FOR REFERENCE:
```

Comment:
Is it useful to keep this code commented here?

Reply:
No, I'm going to remove it.